cmd/golangBuild.go

```diff
@@ -35,6 +35,8 @@ const (
 	golangTestsumPackage        = "gotest.tools/gotestsum@latest"
 	golangCycloneDXPackage      = "github.com/CycloneDX/cyclonedx-gomod/cmd/cyclonedx-gomod@latest"
 	sbomFilename                = "bom.xml"
+	golangciLintCurlURL         = "https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh"
```

Comment:
how would that work on a Windows machine?
Could we find a way to download and install in an OS agnostic way?

Reply:
Added usage of ```go install``` instead of ```curl``` and ```sh``` - [829b6c2](https://github.com/SAP/jenkins-library/pull/3903/commits/829b6c2426fc534021b8c9e2fb8b536b385f28f0)

Reply:
Added OS agnostic way of downloading via ```http``` package.